Content.Client/Light/Components/ExpendableLightComponent.cs

```diff
@@ -1,14 +1,49 @@
+using Content.Client.Light.EntitySystems;
 using Content.Shared.Light.Component;
 using Robust.Shared.Audio;
 
-namespace Content.Client.Light.Components
+namespace Content.Client.Light.Components;
+
+/// <summary>
+/// Component that represents a handheld expendable light which can be activated and eventually dies over time.
+/// </summary>
+[RegisterComponent]
+public sealed class ExpendableLightComponent : SharedExpendableLightComponent
 {
     /// <summary>
-    ///     Component that represents a handheld expendable light which can be activated and eventually dies over time.
+    /// The icon state used by expendable lights when the they have been completely expended.
+    /// </summary>
+    [DataField("iconStateSpent")]
+    public string? IconStateSpent;
+
+    /// <summary>
+    /// The icon state used by expendable lights while they are lit.
+    /// </summary>
+    [DataField("iconStateLit")]
+    public string? IconStateLit;
+
+    /// <summary>
+    /// The sprite layer shader used while the expendable light is lit.
+    /// </summary>
+    [DataField("spriteShaderLit")]
+    public string? SpriteShaderLit = "shaded";
+
+    /// <summary>
+    /// The sprite layer shader used after the expendable light has burnt out.
+    /// </summary>
+    [DataField("spriteShaderSpent")]
+    public string SpriteShaderSpent = "shaded";
```

Comment:
There is no "shaded" shader, i see where the confusion comes from since some gen visualizers uses it but thats because their shader cant be reset to null since there is a null check in their internals, so they get set to a non-existant one (shaded) as a hack, but thats not necessary here since sprites are """"shaded"""" by default

Reply:
Alright.